generateSql.js

```diff
@@ -0,0 +1,99 @@
+// This is a utility file to generate SQL INSERT statments
+// for the schema.sql file. **You don't need to use this.**
+// To actually seed your database, please see seed-database
+// script in package.json
+
+const loremIpsum = require('lorem-ipsum');
+
+// INSERT INTO trail (trail_id, trail_name, distance,
+//   distance_units, elevation_gain, elevation_units,
+//   description, route_type, difficulty, general_area, origin)
+//   VALUES (20, "Wiliwilinui Bernie Ridge Trail", 4.5, "miles",
+//   1597, "ft", "Wiliwilinui Ridge Trail is a 4.5 mile out and
+//   back trail located near Honolulu, Hawaii that offers the
+//   chance to see wildlife and is rated as moderate. The trail
+//   offers a number of activity options and is accessible
+//   year-round. Dogs are also able to use this trail but must be
+//   kept on leash.", "Out & Back", "Moderate", "Honolulu,
+//   Hawaii",
+//   "https://www.alltrails.com/trail/us/hawaii/wiliwilinui-ridge-trail");
+
+const sentences = function (num, words, len) {
+  let sent = loremIpsum({
+    count: num,
+    units: 'sentences',
+    sentenceLowerBound: 2,
+    sentenceUpperBound: words,
+    format: 'plain'
+  });
+  return sent.substring(0, len);
+};
+
+const words = function (num) {
+  return loremIpsum({
+    count: num,
+    units: 'words',
+    format: 'plain'
+  });
+};
+
+const setNumber = function (max) {
+  return Math.floor(Math.random() * Math.floor(max));
+};
+
+const setOrigin = function () {
+  let domain = loremIpsum({
+    count: 1,
+    units: 'words',
+    format: 'plain'
+  });
+  let sub = loremIpsum({
+    count: 1,
+    units: 'words',
+    format: 'plain'
+  });
+  let page = loremIpsum({
+    count: 1,
+    units: 'words',
+    format: 'plain'
+  });
+  let results = `http://www.${domain}.com/${sub}/${page}.html`;
+  return results.substring(0, 99);
+};
+
+const setRouteType = function () {
+  let types = ['Out & Back', 'Loop', 'Bernie'];
```

Comment:
for greater flexibility, for the fields `routeType` + `difficulty` I would maybe revise to be enumerations and set the descriptor text in your code. so storing a 0, 1, 2... etc for more flexibility and in your code having `routeType.OUT_AND_BACK = 0` etc as an example.

Reply:
Cool, thanks Chris! I'll def check it out & play with that syntax a bit. For this repo, not sure that flexibility is needed since no one will use this `generateSql.js` file, even on local installs. This was just a set of helper functions I created for myself to generate INSERT statements for the `schema.sql` file.

To actually seed a local database, users will just `npm run seed-database` so no need to use this file or those functions.

Reply:
right for the routeType and difficulty your field type is a string. This means if you ever wanted to change `bernieBuster` to `bernieIsABuster` or `Out & Back` to `Out + Back` you'll have to adjust it for ever entry.. Here would be a great opportunity to use an enumeration aka using a  `int` field type instead. minor details though, let me know if you disagrree.



``` js

{
 'OUT_AND_BACK': 0,
'LOOP': 1,
...
}

```

Reply:
Hmm. Okay, sounds interesting, not sure I'm following tho, sorry. `routeType` and `difficulty` are set on lines 65 and 71 https://github.com/rpt09-scully/trail-service/blob/generateData/generateSql.js#L65-L71 so only in one place ever. Not sure how an object with those properties changes that, but I'm prolly missing a big piece of this. I totally ge the enum thing with lots of strings to avoid typos, etc. Just wondering how it differs to have a few object properties vs array values?

Reply:
Okay again its minor, not a major functionality issue, but not talking about lines specifically but actually questioning the types for this https://github.com/rpt09-scully/trail-service/blob/7c76096d0a4729179d8db20728adddc17cf08abf/schema.sql#L15-L16. I would revise to 

### schema
```
  route_type int(2),
  difficulty int(2),
```

### database/index.js

Then in your index.js you store a hashtable to declare some sort of enumeration. 

``` js
const difficulty = {
   'EASY':  0,
   'MODERATE: 1,
   'DIFFICULT': 2,
}

// example query for easy routes (via orm or however...)
connection.query('SELECT * trails WHERE difficulty = ?', [difficulty.EASY])

// for api one can backfill with descriptive text if desired
```

Question: why do this?
 - Less redundant storage of strings = performance + abstraction
 - flexibility with descriptions (change on server rather than db if ever need to for clarity)
 - abstract numerics keeps potential bugs to minimum (strings means case sensitivity , white space, etc, spelling)


Again if this is not desired, then we can just merge. Just maybe i wasn't clear with why/what/how the thing i was talking about lol!